backend/app/sql/shatter_parent.sql

```diff
@@ -0,0 +1,50 @@
+CREATE OR REPLACE FUNCTION shatter_parent(
+    input_document_id UUID,
+    parent_geoids VARCHAR[]
+)
+RETURNS TABLE (
+    output_document_id UUID,
+    output_child_path VARCHAR,
+    output_zone INTEGER
+) AS $$
+DECLARE
+    districtr_map_uuid UUID;
+
+BEGIN
+    SELECT districtrmap.uuid INTO districtr_map_uuid
+    FROM document.document
+    INNER JOIN districtrmap
+    ON document.gerrydb_table = districtrmap.gerrydb_table_name
+    WHERE document.document_id = $1;
+
+    IF districtr_map_uuid IS NULL THEN
+        RAISE EXCEPTION 'District map uuid not found for document_id: %', input_document_id;
+    END IF;
+
+    RETURN QUERY
+    WITH inserted_child_geoids AS (
+        INSERT INTO document.assignments (document_id, geo_id, zone)
+        SELECT $1, child_geoids.child_path, child_geoids.zone
+        FROM (
+            SELECT a.document_id, edges.child_path, a.zone
+            FROM document.assignments a
+            INNER JOIN parentchildedges edges
+            ON edges.parent_path = a.geo_id
+            WHERE a.document_id = $1
+                AND a.geo_id = ANY(parent_geoids)
```

Comment:
(o) might try scooting this into the join condition- I don't have strong intuition if it'll be a a better execution plan, definitely might be.

Reply:
I'll try running `EXPLAIN ANALYZE`!